wpe/src/main/cpp/browser/entrypoints.cpp

```diff
@@ -193,22 +192,65 @@ void updateAllPageSettings(JNIEnv* env, jclass, jint pageId, jobject jPageSettin
 
     Browser::getInstance().updateAllPageSettings(pageId, settings);
 }
-} // namespace
 
-JNIEXPORT void JNICALL wpe_android_launchProcess(uint64_t pid, wpe::android::ProcessType processType, int* fd)
+// Process management API
+
+struct AndroidProcessProvider {
+    struct wpe_process_provider *wpe_provider;
+};
+
+void* createProcessProvider(struct wpe_process_provider *process_provider)
+{
+    ALOGV("BrowserGlue createProcessProvider()");
+    auto *provider = new AndroidProcessProvider { process_provider };
+    return provider;
+}
+
+void destroyProcessProvider(void* data)
 {
-    if (fd == nullptr) {
-        ALOGE("Cannot launch process (invalid file descriptor)");
+    ALOGV("BrowserGlue destroyProcessProvider()");
+    auto *provider = reinterpret_cast<AndroidProcessProvider*>(data);
+    if (!provider)
         return;
-    }
 
+    delete provider;
+}
+
+int32_t launchProcess(void* data, enum wpe_process_type wpeProcessType, void* options)
+{
+    ALOGV("BrowserGlue launchProcess()");
+    auto *provider = reinterpret_cast<AndroidProcessProvider*>(data);
+    if (!provider)
+        return -1;
+
+    auto **argv = reinterpret_cast<char**>(options);
+    if (!argv)
+        return -1;
+
+    uint64_t pid =  std::strtoll(argv[0], nullptr, 10);
+    int32_t fd =  std::stoi(argv[1]);
+
+    auto processTypeToAndroidProcessType = [] (enum wpe_process_type wpeProcessType)
```

Comment:
This could be `static auto`, given that the lambda has no captured variables. One can assume that the compiler should be smart enough to notice, but my preference is to explicitly write it, to make the code a bit self-documenting. Not a strong preference, though; so feel free to leave this as-is.